File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/MethodCallRequest.java
package org.opcfoundation.webserver.types.message;

import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.types.UaObjectId;

import java.util.List;

public class MethodCallRequest {
    private final UaObjectId objectId;
    private final String methodName;
    private final List<Variant> inputArguments;

    public MethodCallRequest(
            UaObjectId objectId,
            String methodName,
            List<Variant> inputArguments)
    {
        this.objectId = objectId;
        this.methodName = methodName;
        this.inputArguments = inputArguments;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public String getMethodName() {
        return methodName;
    }

    public List<Variant> getInputArguments() {
        return inputArguments;
    }
}

File: opcua-webservice-node/source/src/server/service/message/reactiveobject/MethodCallRequest.ts
import { UaVariant } from "opcua-webapi-ts";
import { UaObjectId } from "../../../types";

export class MethodCallRequest {
    private readonly _objectId: UaObjectId;
    private readonly _methodName: string;
    private readonly _inputArguments: Array<UaVariant>;

    constructor(
        objectId: UaObjectId,
        methodName: string,
        inputArguments: Array<UaVariant>,
    ) {
        this._objectId = objectId;
        this._methodName = methodName;
        this._inputArguments = inputArguments;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get methodName(): string {
        return this._methodName;
    }

    get inputArguments(): Array<UaVariant> {
        return this._inputArguments;
    }
}